server/src/project/project.service.ts

```diff
@@ -395,9 +411,26 @@ export class ProjectService {
 
     transformedProject.video_links = await getVideoLinks(this.config.get('AIRTABLE_API_KEY'), firstProject.dcp_name);
 
+    let { records: projectPackages } = await this.crmService.get('dcp_packages', `
+        $filter=
+          _dcp_project_value eq ${firstProject.dcp_projectid}
+          and (
+            dcp_visibility eq ${PACKAGE_VISIBILITY.GENERAL_PUBLIC}
+          )
+          and (
+            statuscode eq ${PACKAGE_STATUSCODE.SUBMITTED}
+          )
+      `);
+
+    projectPackages = await Promise.all(projectPackages.map(async (pkg) => {
+        return await this.documentService.packageWithDocuments(pkg);
+      }));
```

Comment:
How does this perform? I think it's probably okay, but I could see these getting really slow. Have you considered getting the packages in the very first request? We're removing the expansion for keyword entities anyway, so there'll be room for another one.

I wonder if there's a way to get a filtered list of folders in SharePoint based on project name, and also get a list of files there, in a single request? I wouldn't spin your wheels on it too much, but I have a hard time seeing this perform very well. Another thing we could do to prevent a bottle neck is to make the entire files/artifacts section asynchronous from the front end: so, have another component make a separate request to the API to a dedicated endpoint that does this separately.

Reply:
Hmm. Good idea, conceptually speaking including it in the initial expand might be faster.

It's been pretty speedy so far. I think that's because we are only loading document metadata (name, url, etc. instead of the documents themselves). I've only been testing on projects with 1-2 packages and 1-2 documents per package, but I think in terms of absolute response time, even with a dozen packages and dozen documents shouldn't decrease performance too dramatically since it's mostly text results. 

Interesting idea about getting a flat list of Project files from Sharepoint. After this PR I have to rework this code to read the sharepoint url from the package, anyway. I can look into that sort of thing..

Reply:
>  I think that's because we are only loading document metadata (name, url, etc. instead of the documents themselves).

Yep, I know. But it's still a new network request for every package.

Just keep in mind there is a rate limit in Microsoft's API so we'd wanna be mindful. I could see how much we're eating up with the public-facing site.